src/components/SectionNotification/SectionNotification.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { SECTION_NOTIFICATION_DATA_HOOKS } from './dataHooks';
+import {
+  NOTIFICATION_TYPE,
+  SectionNotificationDefaultProps,
+  SectionNotificationProps,
+} from './types';
+import { SectionNotificationButton } from './Button/SectionNotificationButton';
+import { SectionNotificationText } from './Text/SectionNotificationText';
+import { SectionNotificationIcon } from './Icon/SectionNotificationIcon';
+import styles from './SectionNotification.st.css';
+
+/** Section notification displays an important, succinct message, and provides actions for users to address and can not be dismissed.  */
+export class SectionNotification extends React.Component<
+  SectionNotificationProps
+> {
+  static displayName = 'SectionNotification';
+  static defaultProps: SectionNotificationDefaultProps = {
+    type: NOTIFICATION_TYPE.default,
+    'data-hook': SECTION_NOTIFICATION_DATA_HOOKS.root,
+  };
+
+  static Text = SectionNotificationText;
+  static Icon = SectionNotificationIcon;
+  static Button = SectionNotificationButton;
+
+  render() {
+    const { children, type } = this.props;
+    const isError = type === NOTIFICATION_TYPE.error;
+
+    const contents = [];
+    const buttons = [];
+
+    React.Children.forEach(children, (child: React.ReactElement) => {
+      if (child && child.type === SectionNotificationButton) {
+        buttons.push(child);
+      } else {
+        contents.push(child);
+      }
+    });
+
+    return (
+      <div
+        {...styles('root', {
+          error: isError,
+        })}
+        data-error={isError}
+        data-hook={this.props['data-hook']}
```

Comment:
the `data-hook` attribute isn't needed here, since stylable handles this for us.
(that's why we need to pass `this.props` to the stylable class)